File: FinanceDemo/FinanceDemo/Features/Input/InputViewController.m
//
//  InputViewController.m
//  FinanceDemo
//
//  Created by Hyper on 2021/4/30.
//

#import "InputViewController.h"

@interface InputViewController () <UITextFieldDelegate>

@end

@implementation InputViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.textField.delegate = self;
    [self.textField addTarget:self action:@selector(textFieldEditingChanged:) forControlEvents:UIControlEventEditingChanged];
}

- (void)textFieldEditingChanged:(UITextField *)textField {
    [FinanceTools textField:textField fullLength:self.model.fullLength];
}

- (BOOL)textField:(UITextField *)textField shouldChangeCharactersInRange:(NSRange)range replacementString:(NSString *)string {
    return [FinanceTools textField:textField digits:self.model.digits replacementString:string];
}

@end

File: FinanceDemo/FinanceDemo/Classes/Demo/Input/InputViewController.swift
//
//  InputViewController.swift
//  FinanceDemo
//
//  Created by Hyper on 2020/12/18.
//

import UIKit

class InputViewController: ViewController {
    private let label: UILabel = {
        let label = UILabel(frame: .zero)
        label.textColor = .blue
        label.font = .systemFont(ofSize: 22, weight: .medium)
        label.textAlignment = .center
        return label
    }()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        textField.delegate = self
        textField.addTarget(self, action: #selector(textFieldDidChange(textField:)), for: .editingChanged)
    }
}

extension InputViewController: UITextFieldDelegate {
    @objc func textFieldDidChange(textField: UITextField) -> Void {
        label.text = textField.text
    }
}
